src/ndpp_init.F90

```diff
@@ -0,0 +1,622 @@
+    type(Ndpp), pointer  :: ref  => null() ! Current data to point to
+
+    allocate(skip(n_nuclides_total))
+    skip(:) = .False.
+
+    ! Parse through each nuclide to find temperature duplicates
+    ! During this we will make sure the temperatures of ACE and NDPP match,
+    ! as well as making sure the NDPP data exists at all.
+
+    do i_nuclide = 1, n_nuclides_total
+      skip(i_nuclide) = .True.
+      ref => ndpp_nuc_data(i_nuclide)
+      do i_other = 1, n_nuclides_total
+        ! Dont do this one if we already have taken care of it, or it is our
+        ! reference
+        if (skip(i_other)) then
+          cycle
+        end if
+        data => ndpp_nuc_data(i_other)
+        ! Now check to see if zaid is the same as the reference
+        if (ref % zaid == data % zaid) then
+          ! We have a match so just deallocate this guy's temp-indep. data
+          ! and point it to the data of i_nuclide
+          if (associated(data % inel)) then
+            deallocate(data % inel)
+            data % inel => ref % inel
+          end if
+          if (associated(data % nuinel)) then
+            deallocate(data % nuinel)
+            data % nuinel => ref % nuinel
+          end if
+          if (associated(data % inel_Ein)) then
+            deallocate(data % inel_Ein)
+            data % inel_Ein => ref % inel_Ein
+          end if
+          if (associated(data % inel_Ein_srch)) then
+            deallocate(data % inel_Ein_srch)
+            data % inel_Ein_srch => ref % inel_Ein_srch
+          end if
+
+          ! Ok, good work team, now mark skip as true so we dont do it again
+          skip(i_other) = .True.
+
+          ! Clean up the pointers
+          nullify(data)
+          nullify(ref)
+        end if
+      end do
+    end do
+
+    deallocate(skip)
+
+  end subroutine remove_temperature_independent_data
+
+!===============================================================================
+! WHICH_DATA looks at the requested tallies and determines which data is needed
+! from the NDPP libraries so that memory utilization is kept to a minimum.
+!===============================================================================
+
+  subroutine which_data(scatt_type, get_scatt, get_nuscatt, get_chi_t, &
+                        get_chi_p, get_chi_d, scatt_order)
+    integer, intent(in)  :: scatt_type  ! Whether or not legendre or tabular data
+    logical, intent(out) :: get_scatt   ! Flag for whether or not to get scatt data
+    logical, intent(out) :: get_nuscatt ! Flag for whether or not to get nuscatt data
+    logical, intent(out) :: get_chi_t   ! Flag for whether or not to get total chi data
+    logical, intent(out) :: get_chi_p   ! Flag for whether or not to get prompt chi data
+    logical, intent(out) :: get_chi_d   ! Flag for whether or not to get delayed chi data
+    integer, intent(out) :: scatt_order ! Number of moments requested in tallies for scatter
```

Comment:
Please get these guys down to 80 cols as well.  I think our convention for this is to indent multiple lines of comments so that all the exclamations line up (even these little inline guys at the top of subprograms)